cmd/singularity/actions_test.go

```diff
@@ -156,6 +183,30 @@ func testSingularityExec(t *testing.T) {
 			}
 		}))
 	}
+
+	// Clean up
+	err = os.Remove(tmpfile.Name())
```

Comment:
Use defer just after the file is created rather than removing later.

See example here: https://golang.org/pkg/io/ioutil/#TempFile